feat: 로그인/회원가입 페이지 및 기능 추가

kimsuyeon0916 opened the pull request:
## 변경 사항

- 로그인 및 회원가입 페이지 UI 스타일링
- 로그인, 아이디 중복확인, 회원가입, 로그아웃 api 함수 및 api 호출 커스텀 훅 추가
- 로그인 및 회원가입 페이지 fetch 적용
- 커스텀 react-hook-form Devtool 공통 컴포넌트 추가
- Button, Label, BottomSheet, ErrorMessage 스타일 변경
- axios instance 추가 설정
- useSignupForm 내 스키마 및 모드 변경
- User 타입 내 프로퍼티 `username` -> `name` 변경

## 리뷰 필요

- `.env` 파일은 노션에 남겨두었습니다. 앞으로 `.env` 파일을 최상위에 추가하시구 실행하시면 됩니다.
- react-hook-form 내의 Devtool을 클라이언트 컴포넌트에 적용하니, hydration error가 나서 해결을 위해 별도의 공통 컴포넌트로 제작하였습니다. 앞으로 해당 컴포넌트 이용하시면 좋을 것 같습니다!
- ~기존의 useSignupForm 에서 validateMode를 onSubmit이면, 페이지 뒤로가기가 존재하지 않는 현재의 디자인 상태에서는 유효성 검사를 제대로 할 수가 없어서 onChange로 변경하였습니다. (때문에 리렌더링이 좀 있습니다..😂)~
- 기존의 useSignupForm 에 적용된 스키마는 비밀번호 확인이 제대로 되지 않는 문제가 있었습니다. 따라서 스키마의 구조를 변경하였지만 또다른 이슈가 생겨서 원인 파악중입니다...🥲
- 피그마와 상이한 부분들을 맞추기 위해 Button, Label, BottomSheet, ErrorMessage 컴포넌트 스타일을 변경하였습니다.
- 회원가입 시 ux를 고려한 유효성 검사 과정을 추가하였습니다.
  - 아이디 중복 확인 커스텀 validation을 추가하였습니다.
  - 매 단계별 존재하는 폼들의 유효성 검사가 통과가 되어야, 버튼이 활성화되도록 하였습니다.
  - 회원가입 완료 이후, 한번 터치(클릭)하면 로그인 페이지로 이동하도록 하였습니다.
- 현재 토큰을 로컬스토리지에 보관합니다. 추후 보안을 위해 secureStorage로 적용을 하거나 cookie에 저장하는 방식으로 변경 고려중입니다.
- 폴더 구조 및 api 관련 코드 작성에 대한 자유로운 의견 및 질문은 언제나 환영입니다!🙌🏻
  - features 폴더는 라우팅 구조와 동일하게 관심사별로 분리하였습니다. 
  - api와 커스텀 훅 작성 및 사용 방법은 `api/auth`, `hooks/api/useAuth` 참고 부탁드립니다.
    - 로그인, 아이디 중복 확인, 회원가입 api 함수는 헤더에 엑세스 토큰을 지정하지 않아야 하므로 기본 axios http 메서드를 이용하였습니다. 이외에는 axios interceptor(`import { api } from '@/api'`) 사용하시면 됩니다.
- 가족 프로필 조회 및 리프레쉬 토큰 재발급 api 연동은 PR 단위가 커질 것 같아 해당 PR 이후에 적용할 예정입니다. 
  - 현재 리프레시 토큰이 설정되지 않는 이슈가 있습니다. 원인 파악 및 해결 후 다음 PR 에 함께 적용하여 올리겠습니다.
- 서버분들께 알림(personal, notification) 관련해서 api 명세 변경을 요청드릴 예정입니다.
- 디자이너분들께는 로그인 페이지 내 유효성 UI 및 회원가입 내 뒤로가기 버튼을 요청 드릴 예정입니다.
- (추가) pwa로 인해 콘솔에 warning 이 떠서 pwa는 production 모드에서만 적용되도록 설정하였습니다.
  
제가 놓친게 있으면 말씀 부탁드립니다! 
(이번에 PR 단위가 큽니다.. 양해 부탁드립니다..🥺)

### 첨부 링크(옵션)

- [env 파일 정보](https://www.notion.so/c4c870e927e34d2e847cd58698f9c73f?pvs=4)
- [react-hook-form Devtool hydration error 관련 자료](https://github.com/react-hook-form/devtools/issues/187)
- [zod 스키마 관련 자료](https://github.com/colinhacks/zod/issues/479#issuecomment-1536233005)
- [pwa 관련 자료](https://github.com/quasarframework/quasar/issues/7272)

close #93

kimsuyeon0916 committed:
chore: gitignore 추가 (#93)

- .env 추가

kimsuyeon0916 committed:
fix: Button 컴포넌트 변경 (#93)

- disable 시 스타일 지정

kimsuyeon0916 committed:
feat: 이용약관 관련 상수 추가 (#93)

kimsuyeon0916 committed:
feat: TosStep 컴포넌트 추가 (#93)

kimsuyeon0916 committed:
feat: 약관동의 페이지 추가 (#93)

kimsuyeon0916 committed:
feat: CompleteStep 컴포넌트 추가 (#93)

kimsuyeon0916 committed:
feat: 회원가입 완료 페이지 추가 (#93)

kimsuyeon0916 committed:
feat: SignUpFormProvider 컴포넌트 추가 (#93)

kimsuyeon0916 committed:
feat: 회원가입 페이지 layout 추가 (#93)

kimsuyeon0916 committed:
feat: 회원가입 완료 및 로그인 로고 추가 (#93)

kimsuyeon0916 committed:
feat: react-hook-form Devtool 컴포넌트 추가 (#93)

- hydration error 로 인해 별도로 만듦

kimsuyeon0916 committed:
feat: AccountStep 컴포넌트 추가 (#93)

kimsuyeon0916 committed:
feat: UserInfoStep 컴포넌트 추가 (#93)

kimsuyeon0916 committed:
feat: 회원가입 시 로그인 정보 입력 페이지 추가 (#93)

kimsuyeon0916 committed:
feat: 회원가입 시 유저 정보 입력 페이지 추가 (#93)

kimsuyeon0916 committed:
feat: SignInStep 컴포넌트 추가 (#93)

kimsuyeon0916 committed:
feat: 로그인 페이지 추가 (#93)

kimsuyeon0916 committed:
feat: User 타입 변경 (#93)

- username -> name 프로퍼티명 변경
- familyCode 프로퍼티 추가

kimsuyeon0916 committed:
feat: auth 관련 타입 추가 (#93)

- AuthToken
- SignInRequest
- SignInResponse
- SignUpRequest

kimsuyeon0916 committed:
refactor: User 타입 내 username -> name 프로퍼티명 변경 반영 (#93)

kimsuyeon0916 committed:
feat: useSignupForm 변경 (#93)

- 비밀번호 확인 관련 스키마 별도 구성 (비밀번호 확인 유효성 검사가 되지 않는 문제를 해결하기 위함)
- 스키마 구조 변경
- 'onSubmit' -> 'onChange' 모드 변경

kimsuyeon0916 committed:
feat: axios instance 설정 변경 (#93)

- onRequest 헤더에 토큰 설정 과정 추가
- http 메서드 매개변수 타입 제네릭으로 설정

kimsuyeon0916 committed:
feat: 로그인, 회원가입, 아이디 중복 확인, 로그아웃 api 함수 추가 (#93)

kimsuyeon0916 committed:
chore: 빈 gitkeep 파일 제거 (#93)

kimsuyeon0916 committed:
feat: 로그인, 회원가입, 아이디 중복확인, 로그아웃 api 호출 커스텀 훅 추가 (#93)

kimsuyeon0916 committed:
feat: BottomSheet 컴포넌트 내 bgStyle props 추가 (#93)

kimsuyeon0916 committed:
style: Label 컴포넌트 스타일 변경 (#93)

kimsuyeon0916 committed:
feat: 로그인 fetch 적용 (#93)

kimsuyeon0916 committed:
feat: 아이디 중복확인 fetch 적용 (#93)

kimsuyeon0916 committed:
feat: 회원가입 fetch 적용 (#93)

kimsuyeon0916 committed:
style: ErrorMessage 컴포넌트 스타일 추가 (#93)

- mx-[8px] 추가

kimsuyeon0916 committed:
feat: 아이디 중복확인, 회원가입 api 함수 변경 (#93)

- axios interceptor -> axios 기본 http 메서드 사용

kimsuyeon0916 committed:
feat: axios interceptor 헤더 기본 설정 (#93)

- Content-Type을 application/json 으로 지정

kimsuyeon0916 committed:
feat: 로그아웃 api 호출 커스텀 훅 변경 (#93)

- 로컬스토리지 비우는 과정 추가

kimsuyeon0916 committed:
feat: useAuth 중복 타입 import 제거 (#93)

kimsuyeon0916 committed:
chore: js-cookie 추가 (#93)

kimsuyeon0916 committed:
feat: 로그인 api 함수 변경 (#93)

- withCredentials: true 설정 추가
- js-cookie 이용해서 코드 변경

KimDongGyun23 commented on src/features/sign-up/AccountStep.tsx:
```diff
@@ -0,0 +1,97 @@
+  }
+
+  return (
+    <div className="flex-column-between mx-[20px] h-full gap-[32px] bg-white pb-[55px]">
```
< 현재의 문제점 >
높이가 작은 (아이폰 SE 등) 모바일 버전에서 유효성 오류 메세지들이 생길 경우,
버튼의 크기가 줄어듭니다.
![image](https://github.com/user-attachments/assets/2517aa74-2e55-4eb2-a6fd-ff1a78c15819)

<br/>

<문제 원인>
버튼의 상위요소인 `<div>`들을 확인해보면 `height`가 모두 `100%`입니다.
이는 상위요소의 크기를 물려받는 것을 의미하고 위로 계속 올라가다보면 전체 레이아웃이 `height: 100svh`임을 알 수 있습니다.

따라서, 현재 컴포넌트에는 `flex-column`이 적용되어있고, 내부의 높이를 `100svh`로 맞추기 위해 버튼의 높이가 줄어드는 것을 알 수 있습니다.

<br/>

< 해결 방법 >
간단합니다. 상위 요소의 높이를 물려받지 않고, 내부의 크기만큼 자동으로 늘어나면 됩니다. 즉, 현재 `46번째 줄`에서 `h-full`을 제거하면 해결될 것입니다.

kimsuyeon0916 replied:
짚어주셔서 감사합니다! `h-full` 을 제거하면 버튼이 하단에 존재하지 않게 되어서 `h-full` 대신 `min-h-full` 을 주고 회원가입 레이아웃의 최소 높이를 지정하여  해결하였습니다!

KimDongGyun23 commented on src/schema/useSignupForm.ts:
```diff
@@ -21,29 +14,44 @@ const schema = z
-    notification: z.boolean(),
   })
-  .partial()
   .refine((formData) => formData.password === formData.confirm, {
```
`.partial()` 을 `.refine()` 앞에 붙여준다면 비밀번호 검증 오류가 해결될 것입니다.

이는 이전에 비슷한 경험을 겪고 찾아봤었는데, 해당 자료를 찾게된다면 따로 말씀드리겠습니다.

kimsuyeon0916 replied:
기존의 코드에서 `partial()`이 존재하여도 동일하게 적용이 안되어서 제거하였습니당
위의 관련 자료에서 동일한 사례로 인해 `partial()` 대신에 구조를 중첩하여 해결하였길래 동일한 방법을 적용하게 되었습니다..!

그러나 적용 이후에 비밀번호 확인 폼의 유효성 검사는 진행되지만, 비밀번호 -> 비밀번호 확인 동일하게 입력하여도 비밀번호 폼에 남아있는 에러 메세지가 사라지지 않는 새로운 이슈가 생겨 원인 파악중입니다..!🥲

KimDongGyun23 replied:
한편으로 드는 생각이지만 '비밀번호가 일치하지 않습니다'라는 메세지는 `confirm`만의 에러 메세지로 적절해 보입니다. `password` 필드에서는 해당 유효성 검사만 진행하는 것은 어떨까요? 
보통은 해당 메세지가 `confirm` 필드에서만 사용되는 경우를 많이 보아서요.

KimDongGyun23 replied:
제 주관적인 생각으로는 `zod`에서는 `mode: onChange` 상태일때,
현재 입력값이 변경된 필드에 한해서 검사를 진행하는 것 같습니다. 

아이디를 입력하여 유효성 검사를 할 때에 
비밀번호 에러메세지가 안뜬다는 점에서 그렇게 생각했습니다.
실제로 말씀하신 문제상황에서 다시 비밀번호를 올바르게 작성하면 에러메세지가 제거되는 것을 확인할 수 있습니다.

`mode: onChange`를 생각해보면 원하는 필드의 유효성 검사를 실시하기 위해서는
해당하는 필드의 값이 변경되어야만 하는 것 같습니다.


만약 `password`와 `confirm` 모두에게 에러메세지를 보여주고싶다면, `react-hook-form`의 기능들을 사용하여 직접 에러메세지를 주입시켜야한다고 생각합니다.

kimsuyeon0916 replied:
> 제 주관적인 생각으로는 zod에서는 mode: onChange 상태일때, 현재 입력값이 변경된 필드에 한해서 검사를 진행하는 것 같습니다.

맞습니다! 그래서 스키마 구조를 변경한 것도 있었습니다..!

> 한편으로 드는 생각이지만 '비밀번호가 일치하지 않습니다'라는 메세지는 confirm만의 에러 메세지로 적절해 보입니다. password 필드에서는 해당 유효성 검사만 진행하는 것은 어떨까요? 
> 보통은 해당 메세지가 confirm 필드에서만 사용되는 경우를 많이 보아서요.
> 만약 password와 confirm 모두에게 에러메세지를 보여주고싶다면, react-hook-form의 기능들을 사용하여 직접 에러메세지를 주입시켜야한다고 생각합니다.

넵! confirm 필드에서만 메세지가 뜨도록 수정하였습니다!

아래의 논의에 따라 기존의 onChange -> onSubmit 으로 변경하고 다음 버튼 누른 이후에 유효성 검사를 실시하니 이전에말씀드린 추가적인 이슈는 저절로 해결되긴 하네요..! 허헣😅

KimDongGyun23 commented on src/features/sign-up/AccountStep.tsx:
회원가입 로직에 관한 전반적인 생각입니다.
우선 현재 회원가입의 경우 라우팅을 통해 페이지를 이동해가며 절차를 진행하고 있습니다.

<br/>

이에 대한 저의 생각은 다음과 같습니다.
1. 이전 페이지에서 작성한 내용이 다양한 이유 (새로고침과 네트워크 이슈) 들로 인해 날아갈 가능성
2. url을 통한 접속에 대한 처리

이러한 이슈들이 존재하기 때문에 저는 Tab 형태의 방식이 좀 더 적절하지 않을까란 생각이 들었습니다.
Tab 형식의 경우, 새로고침이나 url을 통해 접속하더라도 처음 `AccountStep`부터 다시 시작하기 때문에 이러한 문제들을 방지할 수 있습니다. 

이러한 문제들을 방지할 수 있다면 현재 로직으로도 괜찮습니다.
하지만 서버 측에서나 클라이언트 측에서나 꼬일 수 있는 상황이 꽤 있을 것 같아서 의견 남겨봅니다.

kimsuyeon0916 replied:
해당 부분에 대해서 오랫동안 고민해보았는데요..!
제가 기존에 라우팅을 분리한 이유는 저렇게 회원가입 절차가 ux를 위해 분리된 레퍼런스 사이트들(대표적: google)을 참고하니 라우팅이 분리된 것을 확인할 수 있었습니다. 

> 1. 이전 페이지에서 작성한 내용이 다양한 이유 (새로고침과 네트워크 이슈) 들로 인해 날아갈 가능성

물론 google 은 새로고침을 해도 이전 페이지에서 작성한 내용이 유지가 되었습니다. 이 부분은 기존의 방식을 유지할 경우 [다음의 방법](https://stackoverflow.com/questions/71511930/react-hook-form-and-persistent-data-on-page-reload)을 적용해보려고 합니다

또한 네트워크 이슈가 발생한 경우 tanstack-query 에서 `pending` 상태가 일어나는데 이는 Tab형식의 방식에서도 동일한 문제가 발생하고 더불어 모든 기능 개발에서 해결해야할 이슈라 생각하여 좀 다같이 논의가 필요할 듯 합니다!

주관적인 의견으로는 전역으로 queryClient의 `defaultOptions`에서 [네트워크 모드](https://tanstack.com/query/latest/docs/framework/react/guides/network-mode#network-mode-offlinefirst)를 설정하여 해결하는걸 고려하고 있습니다.

> 2. url을 통한 접속에 대한 처리

google 의 경우에도 페이지 접속을 허용하나, 캐시나 쿠키등에서 기존의 입력했던 값을 기억하여 회원가입 로직을 처리하는 듯 합니다.

해당 부분은 저도 문제점이 있다고 생각하여 `onSubmit` 시에 모든 필드의 값이 존재하는지 확인을 시켜서,
아닌 경우에 뒤로가기를 하여 필드의 값을 입력할 수 있다면 해당 문제를 해결할 수 있지 않을까라는 생각으로 피그마에 디자이너 분들께 뒤로가기 버튼을 요청드렸었습니다..!

제 생각은 위와 같습니다! 혹시 이러한 생각에 대해서 동균님의 의견이 궁금합니다!
(논의 결론에 맞춰서 로직을 새롭게 변경하거나 문제를 해결하도록 하겠습니다!)

KimDongGyun23 replied:
우선 고생 많으셨습니다. 모두 확인해보았고 좋은 방식인 것 같습니다.

한가지 여쭤보고싶은게 점이 있습니다.
로컬에 저장하게 된다면 암호화하여 저장하실 계획이신가요? 아무래도 개인정보이다보니 염려되어서요.

고생하셨습니다.

kimsuyeon0916 replied:
rhf 공식문서에서 제안하는 lsm 을 이용하여 세션 스토리지에 이전에 작성했던 내용을 저장하는 방식으로 리팩토링 해보았습니다! 위의 자료의 다른 방식들은 더 간단한 구현이 가능했으나, 리렌더링이 심한 단점이 있어서 해당 방식으로 구현하였습니다!

회원가입 완료 이후에 세션 스토리지를 제거하고 있긴 하나, 추후 암호화 과정이 필요하다고 생각이 들면 로그인 방식과 함께 리팩토링 하도록 하겠습니다~!

확인해보시고 코멘트 주세요~!

KimDongGyun23 replied:
네 확인했습니다. 고생 많으셨어요👍

KimDongGyun23 commented on src/schema/useSignupForm.ts:
```diff
@@ -21,29 +14,44 @@ const schema = z
   const formMethod = useForm<SignupFormType>({
-    mode: 'onSubmit',
-    reValidateMode: 'onSubmit',
+    mode: 'onChange',
```
제가 생각했던 방식은 다음과 같습니다.

<br/>

우선, `placeholder`를 통하여 입력할 정보들의 힌트를 제공하고 있습니다.
그렇기에 `ErrorMessage`들을 굳이 즉각적으로 보여주어야 하는가? 라는 생각을 했었습니다. 

<br/>

또한, 렌더링의 문제도 심각할 것이라 생각했습니다. 입력 폼이 많기 때문에 크게 다가오는 문제라고 생각했습니다.

<br/>

따라서, `다음 버튼`을 클릭했을 경우 해당 페이지의 유효성 검사를 진행하고,
오류가 있다면 보여주는 방식을 생각했었습니다.

kimsuyeon0916 replied:
저도 리렌더링의 문제가 고민이 되었는데요..!

그러나 버튼 `type= submit` 이어야 해당 유효성 검사가 진행되지 않나요?! 그렇게 되면 FormProvider 내에서 폼을 제출하기 전에 제출이 될까봐 mode를 변경한 것도 있긴 합니다..!

KimDongGyun23 replied:
`react-hook-form`의 `trigger`를 사용한다면 원하는 때에 원하는 필드 값들만 유효성 검사를 할 수 있습니다!

`react-hook-form` 공식문서 : https://react-hook-form.com/docs/useform/trigger

kimsuyeon0916 replied:
그럼 해당 부분을 유효성 검사할땐 폼에 값이 1개라도 없을때만 버튼 비활성화되고, 활성화 된 이후에 버튼을 누르면 유효성 검사가 실행되도록 리팩토링 해볼까요?!

KimDongGyun23 replied:
넵 좋습니다! 부탁드립니다~ :)

kimsuyeon0916 replied:
위의 로직대로 리팩토링 해보았습니다! 확인해보시고 자유롭게 코멘트 주세용!🙌🏻

kimsuyeon0916 replied:
위의 로직대로 리팩토링 해보았습니다! 확인해보시고 자유롭게 코멘트 주세용!🙌🏻

그러나 제가 리팩토링 하면서 고민이 생겼는데요, 해당 로직대로 과정을 수행할 때 값을 올바르게 변경해도 다시 버튼을 누르기 전까지는 유효성 메세지가 남겨져있는 상태이기 때문에 ux 측면에서 좋은지는 의문이 들긴 합니다..!

KimDongGyun23 replied:
제가 일중이라 실행 동작을 확인하지 못해서요,,, confirm에만 에러메세지를 보이게 한다면 정상적으로 동작하지 않나요??? 직접 구현해보았을때 문제점은 없었습니다

kimsuyeon0916 replied:
비밀번호 확인 말씀하시는걸까요? 그 부분은 위의 코멘트에서 자세히 답변 달았습니다!

제가 말씀드린 부분은 다음 버튼을 누르고 유효성 검사가 시행된 이후 상황을 말씀드리는 겁니당..!

KimDongGyun23 commented on src/features/sign-up/AccountStep.tsx:
```diff
@@ -0,0 +1,97 @@
+export const AccountStep = () => {
+  const { watch, getValues, getFieldState, setError, clearErrors } = useFormContext()
+  const isEmpty = !watch('id') || !watch('password') || !watch('confirm')
+  const invalid =
```
```javascript 
const { trigger } = useFormContext()

const handleGoNext = async () => {
  const isValid = await trigger(['id', 'password', 'confirm'])
  if (isValid) // 다음으로 이동
}
```
trigger를 사용한다면 해당 페이지의 유효성 검사를 바로 실행할 수 있습니다.

만약 Tab 방식으로 변경하게 된다면, 유효성 검사를 진행하고 성공 시에 다음으로 이동하는 로직이 조금 더 간결하게 진행될 수 있을 것 같아요!

kimsuyeon0916 replied:
알려주셔서 감사합니다! 참고하여 리팩토링 해보았습니다~!

KimDongGyun23 commented on src/features/sign-up/CompleteStep.tsx:
```diff
@@ -0,0 +1,16 @@
+
+export const CompleteStep = () => {
+  return (
+    <Link href="/sign-in" className="flex-center size-full bg-mint-4">
```
Link에 replace 속성을 주거나 다른 방식들을 사용하여 회원가입 페이지로 다시 돌아오지 못하도록 하면 좋을 것 같습니다.

kimsuyeon0916 replied:
좋습니다! replace 속성 추가하였습니다~!

kimsuyeon0916 committed:
feat: 회원가입 완료 후, 회원가입 페이지로 뒤로가기 방지 (#93)

- Complete 컴포넌트의 Link 태그 내 replace 속성 추가

kimsuyeon0916 committed:
fix: 회원가입 관련 페이지 레이아웃 설정 변경 (#93)

- 기본 레이아웃 h-full -> min-h-full 변경
- form 태그 스타일링 제거

G0MTENG commented on src/components/bottomSheet/BottomSheet.tsx:
```diff
@@ -11,11 +11,13 @@ import { AnimatePortal } from '../portal/AnimatePortal'
 export const BottomSheet = ({
   onClickScrim,
   isShort = false,
+  bgStyle = 'bg-white',
```
개인적으로는 bgStyle이 어떤 기능을 하는걸까 한 번에 코드를 읽어본 후에 이해가 됐습니다
결국 BottomSheet의 style을 주기 보다는 background color를 설정하기 위한 props 같은데
bgColor = 'white' 그리고 className에서는 bg-${bgColor} 다음과 같이 사용하는건 어떠신지 궁금합니다 !

kimsuyeon0916 replied:
<img width="875" alt="image" src="https://github.com/user-attachments/assets/a3863418-180a-4d9f-a1a9-32bd01b1f9bf">

전에 카톡으로 논의 했던 부분으로 기억하는데, [tailwindcss 관련 공식문서](https://v2.tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support)에서 권장하지 않으므로 저렇게 적용하였습니다!

G0MTENG commented on src/components/button/Button.tsx:
```diff
@@ -18,7 +18,7 @@ export const Button = ({
   type = 'button',
 }: PropsWithChildren<ButtonProps>) => {
-  const color = primary ? 'bg-mint-6 text-white' : 'bg-gray-3 text-gray-7'
+  const color = !disabled ? 'bg-mint-6 text-white' : 'bg-gray-3 text-gray-7'
```
button의 color가 disable에 의해 mint-6에서 gray-3으로 변경되는거로 생각하셔서 고치신거로 이해했습니다 !

<img width="216" alt="스크린샷 2024-09-08 오후 3 37 57" src="https://github.com/user-attachments/assets/ba2f9252-7d25-4142-ae94-538850331572">
다음과 같은 "취소"에서는 disabled가 false이지만, gray-3으로 보여져야 한다고 생각합니다.

```js
const color = primary && !disabled ? 'bg-mint-6 text-white' : 'bg-gray-3 text-gray-7'
```

다음과 같은 코드는 어떨지 의견남겨주세용 !!

kimsuyeon0916 replied:
primary 가 하는 역할이 disabled 와 겹치는 것 같아서 위와 같이 수정하였습니다! 
(primary props 도 제거했어야 했으나, 기존의 코드에서 변경 사항이 많아서 일단은 냅뒀습니다)

혹시 제가 생각치 못한 부분이 있다면 말씀해주세요!

G0MTENG replied:
disabled는 button 클릭을 막는 용도로 만들었습니다. 
위 첨부했던 사진처럼 "취소" 이 버튼은 primary가 아니고, disabled도 아닙니다!

kimsuyeon0916 replied:
이해했습니다! 알려주셔서 감사합니다~ 별도의 disabled 스타일을 추가해보았습니다! 확인해보시고 코멘트주세용~!

kimsuyeon0916 committed:
feat: useSignupForm 변경 (#93)

- 리렌더링문제로 인해 'onChange' -> 'onSubmit' 모드 변경

kimsuyeon0916 committed:
feat: 회원가입 시, 로그인 정보 유효성 검사 로직 변경 (#93)

- useEffect 이용하여 watch 함수로 인한 리렌더링 문제 해결
- 버튼 누른 이후에 유효성 검사 발생시키도록 변경
- 아이디 중복 확인 이후에, 결과와 입력 아이디 둘 다 보관하도록 수정 (검사 이후에, 값을 변경한 경우 다시 검사 유도하기 위함)

kimsuyeon0916 committed:
feat: 회원가입 시, 회원 정보 유효성 검사 로직 변경 (#93)

- useEffect 이용하여 watch 함수로 인한 리렌더링 문제 해결
- 버튼 누른 이후에 유효성 검사 발생시키도록 변경

kimsuyeon0916 committed:
feat: 회원가입 시, 이용약관 유효성 검사 로직 변경 (#93)

- useEffect 이용하여 watch 함수로 인한 리렌더링 문제 해결
- 버튼 누른 이후에 유효성 검사 발생시키도록 변경

kimsuyeon0916 committed:
feat: useSignupForm 변경 (#93)

- 비밀번호 확인에서만 비밀번호 불일치 메세지 뜨도록 변경

kimsuyeon0916 committed:
feat: production 모드에서만 pwa 실행 설정 (#93)

kimsuyeon0916 committed:
chore: little-state-machine 추가 (#93)

kimsuyeon0916 committed:
feat: global.d.ts 파일 추가 (#93)

- lsm의 GlobalState 타입 내 signUp 프로퍼티 추가

kimsuyeon0916 committed:
feat: 로그인 api 함수 반환 타입 제거 (#93)

kimsuyeon0916 committed:
feat: lsm 기반 persistSignUpForm 커스텀 유틸 함수 추가 (#93)

kimsuyeon0916 committed:
feat: 회원가입 api 함수 호출 커스텀 훅 변경 (#93)

- 회원가입 이후, 세션스토리지 비우는 과정 추가

kimsuyeon0916 committed:
feat: lsm StateMachineProvider 설정(#93)

kimsuyeon0916 committed:
feat: 회원가입 시 새로고침해도 이전 step 에서 작성한 내용 유지 (#93)

kimsuyeon0916 committed:
style: Button 컴포넌트 disabled 스타일 추가 (#93)

kimsuyeon0916 committed:
Merge branch 'develop' into feature/#93_auth
